### PR TITLE
core: simplify regionSubTree and regionMap

### PR DESCRIPTION
**Pull request by HunDunDM**
Signed-off-by: HunDunDM <hundundm@gmail.com>

### What problem does this PR solve?

The functions of `regionTree`, `regionSubTree`, and `regionMap` are duplicated.

### What is changed and how it works?

* Remove unnecessary `totalKeys` statistics
* Merge `regionSubTree` into `regionTree`
* Remove the statistical function of `regionMap`

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: server/core/region.go [hunk 1]
```diff
@@ -437,165 +437,47 @@ func (r *RegionInfo) GetReplicationStatus() *replication_modepb.RegionReplicatio
...
 
 // RegionsInfo for export
 type RegionsInfo struct {
 	tree         *regionTree
-	regions      *regionMap                // regionID -> regionInfo
-	leaders      map[uint64]*regionSubTree // storeID -> regionSubTree
-	followers    map[uint64]*regionSubTree // storeID -> regionSubTree
-	learners     map[uint64]*regionSubTree // storeID -> regionSubTree
-	pendingPeers map[uint64]*regionSubTree // storeID -> regionSubTree
+	regions      regionMap              // regionID -> regionInfo
```
Comment (disksing) [inline server/core/region.go:465]: can use map directly

Comment (HunDunDM) [reply server/core/region.go:465]: This is a split PR, and there are other changes to the function of the subsequent regionMap, which will become complicated.

Comment (disksing) [reply server/core/region.go:465]: cool.

**Review by disksing — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
/lgtm

**Pull request merged**